Update instructions for new site configurations to use openmpi@4.1.6

srherbener opened the pull request:
### Summary

This PR changes the instructions for building on new site configurations to use openmpi@4.1.6. openmpi@4.1.6 contains fixes that enable compiling openmpi successfully using apple-clang@15.0.0.

### Testing

Describe the testing done for this PR.

Successfully built openmpi (version 4.1.6) on my M1 MacBook using apple-clang@15.0.0 compilers.

### Applications affected

List all known applications (UFS WM, JEDI, SRW, etc.) intentionally or unintentionally affected by this PR.

This PR does not update instructions for building on HPC platforms so the impact should be limited to apps on the Mac and Linux based laptops.

### Systems affected

List all systems intentionally or unintentionally affected by this PR.
Mac and Linux based laptops.

### Dependencies

- [ ] waiting on https://github.com/JCSDA/spack/pull/360

### Issue(s) addressed

Fixes #829

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

srherbener committed:
Update openblas version to 0.3.24 for apple-clang-15.x

srherbener committed:
Merge remote-tracking branch 'jcsda/develop' into feature/openblas-0.3.24

srherbener committed:
Updated submodule hash for doc/CMakeModules

srherbener committed:
Updated spack submodule hash for

srherbener committed:
Updated spack submodule hash

srherbener committed:
Sync'd up doc/CMakeModules submodule with jcsda/develop

srherbener committed:
Merge remote-tracking branch 'jcsda/develop' into feature/apple-clang-15.x

srherbener committed:
Merge remote-tracking branch 'jcsda/develop' into feature/apple-clang-15.x

srherbener committed:
Updated spack submodule commit hash

srherbener committed:
Merge remote-tracking branch 'jcsda/develop' into feature/openmpi-4.1.6

srherbener committed:
Updated .gitmodules to point the feature/openmpi-4.1.6 branch in spack

srherbener committed:
Updated version number for openmpi to 4.1.6 for macos instructions.

srherbener committed:
Updated documentation for new site config for Linux to use openmpi@4.1.6.
Updated macos CI tests to build and load openmpi@4.1.6

srherbener commented:
I made an attempt to update the CI testing to use openmpi@4.1.6. I'm fairly certain I got the macos tests updated, but not so sure about the ubuntu test. In the ubuntu testing, I didn't see the explicit specification of the openmpi version so if that means it takes the latest available, then we should be set. If not, then there still needs to be an update somewhere to make the ubuntu testing use openmpi@4.1.6.

climbfuji commented:
> I made an attempt to update the CI testing to use openmpi@4.1.6. I'm fairly certain I got the macos tests updated, but not so sure about the ubuntu test. In the ubuntu testing, I didn't see the explicit specification of the openmpi version so if that means it takes the latest available, then we should be set. If not, then there still needs to be an update somewhere to make the ubuntu testing use openmpi@4.1.6.

Thanks for updating the macOS CI tests. I'll double-check, but if the CI builds succeed (which explicitly load `stack-openmpi/4.1.6`) then that's all correct. For ubuntu, we build with intel+intel-oneapi-mpi, therefore no changes needed.

srherbener committed:
Reverted .gitmodules to default state, updated spack submodule commit hash

srherbener commented:
The associated spack PR (jcsda/spack/pull/360) has been merged, so I have reverted the .gitmodules to point to the jcsda/jcsda_emc_spack_stack branch, and updated the spack submodule commit hash accordingly. This PR is ready for review now. Thanks!
